homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,160 @@
+"""Support for Modbus lights."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.light import LightEntity
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_COMMAND_OFF,
+    CONF_COMMAND_ON,
+    CONF_DEVICE_CLASS,
+    CONF_LIGHTS,
+    CONF_NAME,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType
+
+from .base_platform import BasePlatform
+from .const import (
+    CALL_TYPE_COIL,
+    CALL_TYPE_WRITE_COIL,
+    CALL_TYPE_WRITE_REGISTER,
+    CONF_INPUT_TYPE,
+    CONF_STATE_OFF,
+    CONF_STATE_ON,
+    CONF_VERIFY,
+    CONF_WRITE_TYPE,
+    MODBUS_DOMAIN,
+)
+from .modbus import ModbusHub
+
+PARALLEL_UPDATES = 1
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config: ConfigType, async_add_entities, discovery_info=None
+):
+    """Read configuration and create Modbus lights."""
+    if discovery_info is None:
+        return
+    lights = []
+    for entry in discovery_info[CONF_LIGHTS]:
+        hub: ModbusHub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        lights.append(ModbusLight(hub, entry))
+    async_add_entities(lights)
+
+
+class ModbusLight(BasePlatform, LightEntity, RestoreEntity):
+    """Base class representing a Modbus light."""
+
+    def __init__(self, hub: ModbusHub, config: dict) -> None:
+        """Initialize the light."""
+        config[CONF_INPUT_TYPE] = ""
+        config[CONF_DEVICE_CLASS] = ""
```

Comment:
The device class should be None in this case.
```suggestion
```

Reply:
Done